src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTreeUI.java

```diff
@@ -1334,6 +1334,9 @@ protected void uninstallKeyboardActions() {
      */
     protected void uninstallComponents() {
         if(rendererPane != null) {
+            if (!tree.isVisible()) {
+                rendererPane.removeAll();
```

Comment:
Wouldn't it be enough to remove `rendererPane` from and assign `null` to it?

It's done this way in `BasicTableUI` and `BasicListUI`:
https://github.com/openjdk/jdk/blob/0d8543d6773a516dad54038070dce507179d0709/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java#L1656-L1658
and
https://github.com/openjdk/jdk/blob/0d8543d6773a516dad54038070dce507179d0709/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicListUI.java#L986-L988

Reply:
It is already been done in `completeUIUninstall` called from `uninstallUI`
`rendererPane.removeAll `is done to remove all components that are added in `getNodeDImensions` which gets removed in `paint` but if tree is not visible, then `paint `is not called so those components are not removed, which I am trying to do now in `uninstallUI `by checking tree visibility condition

Reply:
Okay, I see it. Then the added lines in `uninstallComponents` aren't needed because `rendererPane` is assigned `null` in `completeUIUninstall` which is called right after `uninstallComponents`, therefore `rendererPane` itself and all the components added into it are eligible for garbage collection.

Where is the memory leak then?